addon/components/cfb-form-editor/question.js

```diff
@@ -50,11 +52,69 @@ export default Component.extend(ComponentQueryManager, {
     }));
   }),
 
+  prepareSlug() {
+    const $input = $('[name="slug"]');
+    const length = $input.val().length;
+
+    $input.off("keypress.slug keydown.slug");
+    $input.on("keypress.slug keydown.slug", event => {
+      const key = event.key;
+      const shift = event.shiftKey;
+      const input = event.target;
+
+      if (key === "Home") {
+        input.selectionStart = length;
+
+        if (!shift) {
+          input.selectionEnd = length;
+        }
+
+        event.preventDefault();
+      } else if (input.selectionStart < length) {
+        event.preventDefault();
+      } else if (input.selectionStart === length) {
+        if (input.selectionStart === input.selectionEnd) {
+          if (key === "ArrowLeft" || key === "Backspace") {
+            event.preventDefault();
+          }
+        } else {
+          if (key === "ArrowLeft") {
+            if (!shift) {
+              input.selectionEnd = input.selectionStart;
+            }
+            event.preventDefault();
+          } else if (key === "Backspace") {
+            input.setRangeText("", input.selectionStart, input.selectionEnd);
+            event.preventDefault();
+          }
+        }
+      }
+    });
+
+    $input.off("focus.slug click.slug select.slug");
+    $input.on("focus.slug click.slug select.slug", event => {
+      const input = event.target;
+      if (input.selectionStart <= length) {
+        input.selectionStart = length;
+      }
+    });
+  },
```

Comment:
What is this doing? :sweat_smile:

Reply:
Keeping the user from modifying the namespace while still allowing him to edit the rest as expected.

Reply:
I thought about overlaying another element. But that is not as straight forward as I initially thought as the namespace has a variable width which would have to be somehow taken into account. The easiest solution in the end was to have the whole string (namespace and slug) in the same input and restrict modification in the UI.

There's a small bug when selecting text from the first editable character onward. But with normal usage this should not be an immediate issue.

Reply:
Hmm, why not just the low-tech approach?
```html
<div>
  {{namespace}}
  <input type="text" />
</div>
```

Reply:
Hmmm.. Flexbox might make this possible. I'll look into that. With traditional methods there would be no way to size the input element to fill the remaining space after the namespace string.

Reply:
Sure? I'd expect this to work even without flexbox with just a `float`. Anyway, flexbox or the UIkit grid are definitely better.

Reply:
The problem with floating is that you usually get into sizing complications.

But I just stumbled onto another issue. The input field is created by ember-validated-form. Not sure if it is possible to amend the generated markup. But I'll look into that.

Reply:
The way the required field is positioned might offer some guidance:
```html
      <div uk-grid class="uk-grid-small uk-margin">
        <div class="uk-width-expand">
          {{f.input
            type="text"
            label=(t "caluma.form-builder.question.slug")
            name="slug"
            required=true
            disabled=slug
            on-update=(action "updateSlug")
          }}
        </div>

        <div class="uk-width-auto uk-flex">
          {{f.input
            name="isRequired"
            label=(t "caluma.form-builder.question.isRequired")
            required=true
            renderComponent=(component "cfb-jexl-boolean-toggle-switch" size="small")
            class="uk-flex uk-flex-between uk-flex-column"
          }}
        </div>
      </div>
```

Reply:
Something like this?

![namespace-slug](https://i.ibb.co/rc967gN/namespace-slug.png)